packages/flutter_test/test/live_widget_controller_test.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+Future<void> main() async {
+  final List<String> logs = <String>[];
+  runApp(
```

Comment:
Put executions into a `test` body. Code outside of `test` can only prepare stateless data and utility functions.

Reply:
I tried but failed. `runApp` inside a test body will result in `await SchedulerBinding.instance.endOfFrame;` to never complete.

Reply:
Just call `tester.pump` right?

Reply:
@dnfield there isn't and shouldn't be a tester because getting a `WidgetTester` will introduce its own controller, making the `LiveWidgetController`'s signal directed to the wrong place.

Reply:
Sorry I made a mistake. Previous unending await is caused by another reason.

Reply:
Ahhhh ok.  In that case you can just manually drive the binding. But I'm surprised putting this in a test body makes a difference, especially since you're not using testWidgets.

Reply:
@dnfield Sorry the difference I thought it was is caused by some other things.